### DISCUSSION TITLE
LXC backend?

### DISCUSSION THREAD

**Discussion by jnsgruk** [dagger/dagger#2945] [Ideas]
Hey :wave: 

Interested in if folks would find a LXC backend for Dagger useful, and how difficult that would likely be to implement?

I've got a couple of CI pipelines for end-to-end testing of services that check on the state of systemd units, and presence of snap packages etc. These things are pretty hard to achieve in a purely OCI-image world; adding LXC support would still keep the process pretty light and container based, but give more options to teams with more system-oriented code to test?

Many thanks!

Jon

**Comment by samalba** [comment]
Dagger uses [Buildkit](https://github.com/moby/buildkit/) to manage containers. And Buildkit relies on [runc](https://github.com/opencontainers/runc). So you should ask this question on the buildkit repository.

However technically, nothing prevents you to leverage `lxc-*` commands from within dagger actions, you would just have to mount the LXC socket to the containers you run with dagger.

**Comment by jnsgruk** [comment]
Yeh, fair enough - seems like BuildKit itself is probably a little too tied to OCI to make LXD a likely feature.

I have got something working by mounting the socket for now, thanks for the tip.
